tensorboard/webapp/plugins/npmi/util/filter_annotations.ts

```diff
@@ -26,27 +26,31 @@ export function filterAnnotations(
   activeRuns: string[],
   metricArithmetic: ArithmeticElement[],
   metricFilters: MetricFilterListing,
-  metrics: string[]
+  metrics: string[],
+  annotationsRegex: string
 ): AnnotationDataListing {
   const data: AnnotationDataListing = {};
   const allRuns = new Set(activeRuns);
   const allMetrics = new Set(
     metrics.map((metric) => stripMetricString(metric))
   );
   Object.entries(annotationData).forEach((entry) => {
-    let valueDataElements = entry[1];
-    // Remove all inactive runs and keep only metrics currently displayed
-    valueDataElements = valueDataElements.filter((valueDataElement) => {
-      return (
-        allRuns.has(valueDataElement.run) &&
-        allMetrics.has(valueDataElement.metric)
-      );
-    });
-    if (
-      checkArithmeticParts(metricArithmetic, metricFilters, valueDataElements)
-    ) {
-      if (valueDataElements.length !== 0) {
-        data[entry[0]] = valueDataElements;
+    const filterRegex = new RegExp(annotationsRegex, 'i');
+    if (filterRegex.test(entry[0])) {
+      let valueDataElements = entry[1];
+      // Remove all inactive runs and keep only metrics currently displayed
+      valueDataElements = valueDataElements.filter((valueDataElement) => {
+        return (
+          allRuns.has(valueDataElement.run) &&
+          allMetrics.has(valueDataElement.metric)
+        );
+      });
+      if (
+        checkArithmeticParts(metricArithmetic, metricFilters, valueDataElements)
```

Comment:
`checkArithmeticParts` does not read well in my head. Under what criterion is it returning true? Consider refactoring the name to capture the contract.